### PR TITLE
Kitties part-01

### PR DESCRIPTION
**Pull request by WalquerX**
This part of the tutorial covers:

- Setting up node template and get to know how it works.
- Copy and changing the name of the template pallet to use it as scaffold of a custom one.
- Writing the kitties pallet. 
-  Setting up substrate front end template.
- Exploring the template and the polkadot js api via polkadot js console.
- Writing front end components to interact with kitties chain.

### CODE REVIEW

File: tutorials/kitties-part-01/0-intro.mdx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+tags:
+  - pallet
+  - poladot.js
```
Comment (niklasp) [inline tutorials/kitties-part-01/0-intro.mdx:4]:
```suggestion
  - polkadot.js
```

File: tutorials/kitties-part-01/0-intro.mdx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+keywords: [pallet, polkadot-js]
+description:
+  From the substrate node template to a fully functional blockchain that handles kitties NFTs.
+level: intermediate
+duration: 2h
```
Comment (niklasp) [inline tutorials/kitties-part-01/0-intro.mdx:9]: is that realistic?

File: tutorials/kitties-part-01/0-intro.mdx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+---
+
+# How to build the Substrate Kitties Chain
+
+![kitties](./assets/kitties.png)
+
+Welcome to the Substrate Kitties tutorial part I. Designed to teach how to build a blockchain that handles
```
Comment (niklasp) [inline tutorials/kitties-part-01/0-intro.mdx:16]:
```suggestion
Welcome to the Substrate Kitties tutorial part 1. Designed to teach how to build a blockchain that handles 
```

Comment (niklasp) [reply tutorials/kitties-part-01/0-intro.mdx:16]: bettter readable

File: tutorials/kitties-part-01/0-intro.mdx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+![kitties](./assets/kitties.png)
+
+Welcome to the Substrate Kitties tutorial part I. Designed to teach how to build a blockchain that handles 
+the creation and ownership management of NFT Kitties.
+
+It will cover various aspects of building the Kitties pallet, including creation, breeding, 
+transfer, price setting, and purchasing kitties. Additionally, it will provide guidance on developing the 
+front-end UI for interaction with the chain.
+
+This tutorial is an updated/adapted version of two previous tutorials, the [CryptoKitties on Substrate](https://www.shawntabrizi.com/substrate/cryptokitties-on-substrate/)
```
Comment (niklasp) [inline tutorials/kitties-part-01/0-intro.mdx:23]: Maybe add a line of code, why you updated it. is the others outdated? did you add features?

File: tutorials/kitties-part-01/0-intro.mdx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+- Learn patterns for building and running a Substrate node.
+- Write and integrate a custom FRAME pallet to your runtime.
+- Learn how to create and update storage items.
+- Write pallet extrinsics and helper functions.
+- Use the PolkadotJS API to connect to a Substrate node.
+
+## Prerequisites
+
+- Rust knowledge. If you are new to Rust, check the [rust book](https://doc.rust-lang.org/book/).
```
Comment (niklasp) [inline tutorials/kitties-part-01/0-intro.mdx:48]: very nice, with learning links!

File: tutorials/kitties-part-01/2-step2.mdx [hunk 6]
```diff
@@ -0,0 +1,162 @@
+# Renaming the TemplateModule Pallet
+
+We will copy and rename the template pallet so to have it renamed as kitties. This implies making some
```
Comment (niklasp) [inline tutorials/kitties-part-01/2-step2.mdx:3]:
```suggestion
We will copy and rename the template pallet so to have it renamed as `kitties`. This implies making some 
```

File: tutorials/kitties-part-01/2-step2.mdx [hunk 7]
```diff
@@ -0,0 +1,162 @@
+# Renaming the TemplateModule Pallet
+
+We will copy and rename the template pallet so to have it renamed as kitties. This implies making some 
+changes along side the pallet and the runtime.
+
+- Go to the pallets folder and make a duplicate of the template pallet folder and rename it to kitties.
```
Comment (niklasp) [inline tutorials/kitties-part-01/2-step2.mdx:6]:
```suggestion
- Go to the pallets folder and make a duplicate of the template pallet folder and rename it to `kitties`.
```

File: tutorials/kitties-part-01/2-step2.mdx [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+```bash
+cd pallets
+cp -r ./template ./kitties
+```
+- Inside kitties folder modify the cargo.toml file to have `pallet-kitties` as the pallet name.
```
Comment (niklasp) [inline tutorials/kitties-part-01/2-step2.mdx:12]:
```suggestion
- Inside kitties folder modify the `cargo.toml` file to have `pallet-kitties` as the pallet name.
```

File: tutorials/kitties-part-01/2-step2.mdx [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+	);
+}
+```
+- Compile and run it to check that everything is working as expected. Return to the root folder and run:
+
+```bash
+cargo build --release
+./target/release/node-template --dev
+```
+You should be able to check in the [polkadot-js app UI](https://polkadot.js.org/apps/#/explorer)
```
Comment (niklasp) [inline tutorials/kitties-part-01/2-step2.mdx:153]: Add a paragraph how to connect it to the local host?

File: tutorials/kitties-part-01/3-step3.mdx [hunk 10]
```diff
@@ -0,0 +1,1332 @@
...
+Now try running the following command to build your pallet. We won't build the entire chain just yet 
+because we haven't yet implemented the Currency type in our runtime. At least we can check that there are 
+no errors in our pallet so far:
+
+```bash
+cargo build -p pallet-kitties
+```
+We will get some errors related to imports that we have to fix.
+
+* sp_io::hashing::blake2_128:
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:181]:
```suggestion
* `sp_io::hashing::blake2_128`
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 11]
```diff
@@ -0,0 +1,1332 @@
...
+	"codec/std",
+	"frame-benchmarking?/std",
+	"frame-support/std",
+	"frame-system/std",
+	"scale-info/std",
+	"sp-io/std",
+]
+```
+
+* sp_runtime::ArithmeticError
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:215]:
```suggestion
* `sp_runtime::ArithmeticError`
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 12]
```diff
@@ -0,0 +1,1332 @@
...
+— we're going to be using those imports in the later parts of the tutorial.
+
+## Add storage items
+
+Let's start adding the most simple logic we can to our runtime: a function that stores a variable in 
+runtime. In our case that variable wil be the number of kitties in existence.
+
+For any storage item we want to declare, we must include the `#[pallet::storage]` macro beforehand.
+
+In `./pallets/kitties/src/lib.rs`, put the following code to replace "TODO: Storage that keeps track of the
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:258]:
```suggestion
In `./pallets/kitties/src/lib.rs`, put the following code to replace `TODO: Storage that keeps track of the number of kitties in existence`:
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 13]
```diff
@@ -0,0 +1,1332 @@
...
+
+## Add storage items
+
+Let's start adding the most simple logic we can to our runtime: a function that stores a variable in 
+runtime. In our case that variable wil be the number of kitties in existence.
+
+For any storage item we want to declare, we must include the `#[pallet::storage]` macro beforehand.
+
+In `./pallets/kitties/src/lib.rs`, put the following code to replace "TODO: Storage that keeps track of the 
+number of kitties in existence":
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:259]:
```suggestion
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 14]
```diff
@@ -0,0 +1,1332 @@
...
+
+```rust
+/// Keeps track of the number of kitties in existence.
+	#[pallet::storage]
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:263]:
```suggestion
#[pallet::storage]
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 15]
```diff
@@ -0,0 +1,1332 @@
...
+	pub(super) type CountForKitties<T: Config> = StorageValue<_, u64, ValueQuery>;
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:264]:
```suggestion
pub(super) type CountForKitties<T: Config> = StorageValue<_, u64, ValueQuery>;
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 16]
```diff
@@ -0,0 +1,1332 @@
...
+```
+
+This creates a storage item for our pallet to keep track of the total count of Kitties in existence.
+
+## Add Currency implementation
+
+Before we proceed with building our node, we need to add the `Currency` type to our pallet's runtime 
+implementation. In `./runtime/src/lib.rs`, add the following:
+
+```
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:274]:
```suggestion
```rust
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 17]
```diff
@@ -0,0 +1,1332 @@
...
+
+We have just created a struct that requires a custom type called `Gender`. This type will handle an enum 
+defining our Kitty's gender. To create it, you'll build out the following parts:
+
+* An enum declaration, which specifies Male and Female values.
+* Implement a helper function for our Kitty struct.
+
+1. Declare the custom enum
+
+Replace "TODO: Enum that represents the gender of a kitty" with the following code:
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:381]:
```suggestion
Replace `TODO: Enum that represents the gender of a kitty` with the following code:
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 18]
```diff
@@ -0,0 +1,1332 @@
...
+
+```rust
+// Represents the gender of a kitty.
+	#[derive(Clone, Encode, Decode, PartialEq, Copy, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+	// We need this to pass kitty info for genesis configuration
+	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+	pub enum Gender {
+		Male,
+		Female,
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:390]: Add another geneder "Other". we are in 2023 :) and see if other code is affected

File: tutorials/kitties-part-01/3-step3.mdx [hunk 19]
```diff
@@ -0,0 +1,1332 @@
...
+
+2. Implement a helper function for our Kitty struct
+
+Configuring an struct is useful in order to pre-define a value in our struct. For example, when setting a 
+value in relation to what another function returns. In our case we have a similar situation where we need 
+to configure our Kitty struct in such a way that sets `Gender` according to a Kitty's DNA.
+
+We'll create a function called gen_dna that returns a tuple of DNA and gender for a kitty.
+
+Replace "TODO: Generates and returns DNA and Gender for a new kitty" with the following:
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:427]:
```suggestion
Replace `TODO: Generates and returns DNA and Gender for a new kitty` with the following:
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 20]
```diff
@@ -0,0 +1,1332 @@
...
+			}
+		}
+```
+### On-chain randomness
+
+If we want to be able to tell these Kitties apart, we need to start giving them unique properties! In 
+the previous step, we've made use of KittyRandomness which we haven't actually defined yet. Let's get to 
+it.
+
+We'll be using the Randomness trait from frame_support to do this. It will be able to generate a random
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:463]:
```suggestion
We'll be using the Randomness trait from `frame_support` to do this. It will be able to generate a random 
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 21]
```diff
@@ -0,0 +1,1332 @@
...
+In order to use the Randomness trait for our pallet, we must:
+
+1. Define a new type bound by Randomness trait in our pallet's configuration trait:
+
+The `Randomness` trait from `frame_support` requires specifying it with a paramater to replace the 
+`Output` and `BlockNumber` generics. For our purposes, we want the output of functions using this 
+trait to be `Blake2 128-bit hash` which you'll notice should already be declared at the top of your 
+working codebase.
+
+Replace "TODO: The type of Randomness we want to specify for this pallet." with the following:
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:475]:
```suggestion
Replace `TODO: The type of Randomness we want to specify for this pallet.` with the following:
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 22]
```diff
@@ -0,0 +1,1332 @@
...
+Given that we have added a new type in the configuration of our pallet, we need to config our runtime to 
+set its concrete type. This could come in handy if ever we want to change the algorithm that 
+`KittyRandomness` is using, without needing to modify where it's used inside our pallet.
+
+To showcase this point, we're going to set the `KittyRandomness` type to an instance of FRAME's 
+`InsecureRandomnessCollectiveFlip` (It is called insecure because it should not be used in production 
+settings). Conveniently, the Node Template already has an instance of the `InsecureRandomnessCollectiveFlip` 
+pallet. All you need to do is set the `KittyRandomness` type in your runtime, inside `./runtime/src/lib.rs`:
+
+ ```
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:492]:
```suggestion
```rust
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 23]
```diff
@@ -0,0 +1,1332 @@
...
+#[pallet::storage]
+pub(super) type KittiesOwned<T: Config> = StorageMap<
+	_,
+	Twox64Concat,
+	T::AccountId,
+	BoundedVec<[u8; 16], T::MaxKittiesOwned>,
+	ValueQuery,
+>;
+```
+Your turn! Copy the two code snippets above to replace "TODO: Storage that maps the kitty struct to the
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:561]:
```suggestion
Your turn! Copy the two code snippets above to replace `TODO: Storage that maps the kitty struct to the kitty DNA.`
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 24]
```diff
@@ -0,0 +1,1332 @@
...
+pub(super) type KittiesOwned<T: Config> = StorageMap<
+	_,
+	Twox64Concat,
+	T::AccountId,
+	BoundedVec<[u8; 16], T::MaxKittiesOwned>,
+	ValueQuery,
+>;
+```
+Your turn! Copy the two code snippets above to replace "TODO: Storage that maps the kitty struct to the 
+kitty DNA." and "TODO: Storage that tracks the kitties owned by each account.".
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:562]:
```suggestion
 and `TODO: Storage that tracks the kitties owned by each account.`
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 25]
```diff
@@ -0,0 +1,1332 @@
...
+block at execution time.
+
+Substrate's weighting system forces developers to think about the computational complexity each extrinsic 
+carries before it is called so that a node will account for it's worst case, avoiding lagging the 
+network with extrinsics that may take longer than the specified block time. Weights are also 
+intimately linked to the fee system for a signed extrinsic.
+
+For this simple application, we're going to default all weights to 0.
+
+Replace "TODO: Create a new unique kitty." with the following:
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:654]:
```suggestion
Replace `TODO: Create a new unique kitty.` with the following:
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 26]
```diff
@@ -0,0 +1,1332 @@
...
+The `mint()` function will take the following arguments:
+
+* `owner`: of type `&T::AccountId` - this indicates whom the kitty belongs to.
+* `dna`: of type `[u8; 16]` - this specifies the DNA of the kitty going to be minted. If None is passed in, a random DNA will be generated.
+* `gender`: of type `Gender`.
+* `generation`: a `u64` type.
+
+And it will return `Result<[u8; 16], DispatchError>`.
+
+Paste in the following code snippet to write the `mint()` function, replacing "TODO: Mint a new kitty and
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:693]:
```suggestion
Paste in the following code snippet to write the `mint()` function, replacing `TODO: Mint a new kitty and add it to the storage.`
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 27]
```diff
@@ -0,0 +1,1332 @@
...
+
+* `owner`: of type `&T::AccountId` - this indicates whom the kitty belongs to.
+* `dna`: of type `[u8; 16]` - this specifies the DNA of the kitty going to be minted. If None is passed in, a random DNA will be generated.
+* `gender`: of type `Gender`.
+* `generation`: a `u64` type.
+
+And it will return `Result<[u8; 16], DispatchError>`.
+
+Paste in the following code snippet to write the `mint()` function, replacing "TODO: Mint a new kitty and 
+add it to the storage." in the working codebase:
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:694]:
```suggestion
 in the working codebase:
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 28]
```diff
@@ -0,0 +1,1332 @@
...
+
+```rust
+// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+        //--snip--//
+```
+Declare your pallet events by replacing "TODO: Events that can be emitted by the pallet." with:
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:785]:
```suggestion
Declare your pallet events by replacing `TODO: Events that can be emitted by the pallet.` with:
```

File: tutorials/kitties-part-01/3-step3.mdx [hunk 29]
```diff
@@ -0,0 +1,1332 @@
...
+
+    Ok(())
+}
+```
+The `do_transfer` function is more complex than is necessary for a transfer. This is because it is already
+dealing with the logic of a sale. We'll be using this function again in the next extrinsic.
+
+### Buy
+
+Replace "TODO: Buy a kitty that is listed for sale." with:
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:1081]:
```suggestion
Replace `TODO: Buy a kitty that is listed for sale.` with:
```

File: tutorials/kitties-part-01/3-step3.mdx
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:1]: this is a really long page, can you maybe split it into parts wherever you think it would make sens?

File: tutorials/kitties-part-01/0-intro.mdx (hunk 2 above)
Comment (WalquerX) [reply tutorials/kitties-part-01/0-intro.mdx:9]: Do you mean the time? well, compiling alone could take like half an hour. It could take more also depending on how much previous experience someone have.

File: tutorials/kitties-part-01/0-intro.mdx [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+  From the substrate node template to a fully functional blockchain that handles kitties NFTs.
+level: intermediate
+duration: 5h
+---
+
+# How to build the Substrate Kitties Chain
+
+![kitties](./assets/kitties.png)
+
+Welcome to the Substrate Kitties tutorial part 1. Designed to teach how to build a blockchain that handles
```
Comment (niklasp) [inline tutorials/kitties-part-01/0-intro.mdx:16]: Remove "part 1"?

File: tutorials/kitties-part-01/0-intro.mdx [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+
+This tutorial is an updated/adapted version of two previous tutorials, the [CryptoKitties on Substrate](https://www.shawntabrizi.com/substrate/cryptokitties-on-substrate/)
+and the [Build the Substrate Kitties Chain](https://doc.deepernetwork.org/tutorials/v3/kitties/pt1/). The current tutorial uses the polkadot-v0.9.39 version of
+the Substrate Node Template. It was also added some extra sections exploring the node template and the polkadot-js javascript console.
+
+## What we will build?
+
+A substrate based blockchain for Kitties that can do the following:
+
+- Be created either by some original source or by being breed using existing Kitties.
```
Comment (niklasp) [inline tutorials/kitties-part-01/0-intro.mdx:31]: What do you mean by "original source" maybe rephrase?

Id put the enumeration in different format so it doesnt start with "Be" but rather

...based blockchain for Kitties, where you can:

- create Kitties ...
- sell kitties at a price ...
- transfer Kitties ...

File: tutorials/kitties-part-01/3-step3.mdx [hunk 32]
```diff
@@ -0,0 +1,296 @@
+# Kitties pallet: scaffold & main logic
+
+Lets check and ennumerate what we want the pallet to be able to do. And order it in a logic and
```
Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:3]:
```suggestion
Lets check and enumerate what we want the pallet to be able to do. And order it in a logic and 
```

Comment (niklasp) [inline tutorials/kitties-part-01/3-step3.mdx:3]:
```suggestion
Lets check and ennumerate what we want the pallet to be able to do. And order it in a logical and 
```

File: tutorials/kitties-part-01/4-step4.mdx [hunk 33]
```diff
@@ -0,0 +1,299 @@
...
+#### Sketching out the types held by our struct
+
+Looking at the items of our struct from above, we can deduce the following types:
+
+* `[u8; 16]` for dna - to use 16 bytes to represent a Kitty's DNA.
+* `BalanceOf` for price - this is a custom type using FRAME's Currency trait.
+* `Gender` for gender - we are going to create this!
+* `u64` for generation.
+
+So we get the following as our struct kitty struct.
```
Comment (niklasp) [inline tutorials/kitties-part-01/4-step4.mdx:43]:
```suggestion
So we get the following as our kitty struct.
```

File: tutorials/kitties-part-01/4-step4.mdx [hunk 34]
```diff
@@ -0,0 +1,299 @@
...
+type KittyRandomness: Randomness<Self::Hash, Self::BlockNumber>;
+```
+2. Specify the actual type in our runtime
+
+Given that we have added a new type in the configuration of our pallet, we need to config our runtime to 
+set its concrete type. This could come in handy if ever we want to change the algorithm that 
+`KittyRandomness` is using, without needing to modify where it's used inside our pallet.
+
+To showcase this point, we're going to set the `KittyRandomness` type to an instance of FRAME's 
+`InsecureRandomnessCollectiveFlip` (It is called insecure because it should not be used in production
```
Comment (niklasp) [inline tutorials/kitties-part-01/4-step4.mdx:195]: Why are you using an insecure type here? Add a comment or use another one?

File: tutorials/kitties-part-01/4-step4.mdx [hunk 35]
```diff
@@ -0,0 +1,299 @@
...
+    Kitty<T>
+>;
+```
+Breaking it down, we declare the storage type and assign a StorageMap that takes:
+
+* The `Twox64Concat` hashing algorithm.
+* A key of type `[u8; 16]`.
+* A value of type `Kitty<T>`.
+
+The `KittiesOwned` storage item is similar except that we'll be using a `BoundedVec` from  `frame_support::pallet_prelude` to keep track of
```
Comment (niklasp) [inline tutorials/kitties-part-01/4-step4.mdx:252]: Maybe add a comment why using a bounded vec here makes sense

File: tutorials/kitties-part-01/5-step5.mdx [hunk 36]
```diff
@@ -0,0 +1,319 @@
...
+
+### Public and private functions
+
+It is important to understand the pallet design decisions we'll be making around 
+coding up our Kitty pallet's minting and ownership management capabilities.
+
+We will be breaking up the "heavy lifting" logics into private helper functions. This improves code 
+readability and reusability.
+
+Lets paint the big picture of what combining dispatchables and helper functions looks like.
```
Comment (niklasp) [inline tutorials/kitties-part-01/5-step5.mdx:23]:
```suggestion
Let's paint the big picture of what combining dispatchables and helper functions looks like.
```

File: tutorials/kitties-part-01/5-step5.mdx [hunk 37]
```diff
@@ -0,0 +1,319 @@
...
+Every dispatchable function must have an associated weight to it. Weights are an important part of 
+developing with Substrate as they provide safe-guards around the amount of computation to fit in a 
+block at execution time.
+
+Substrate's weighting system forces developers to think about the computational complexity each extrinsic 
+carries before it is called so that a node will account for it's worst case, avoiding lagging the 
+network with extrinsics that may take longer than the specified block time. Weights are also 
+intimately linked to the fee system for a signed extrinsic.
+
+For this application, and since we are developing it for learning purposes, we're going to default all weights to 0.
```
Comment (niklasp) [inline tutorials/kitties-part-01/5-step5.mdx:56]: Please add one line on where to find documentation about weight can be calculated.

File: tutorials/kitties-part-01/9-conclusion.mdx [hunk 38]
```diff
@@ -0,0 +1,15 @@
+# Conclusion
+
+We have completed the Substrate Kitties Tutorial. We have learned how to build a blockchain using Substrate and how to create a runtime
```
Comment (niklasp) [inline tutorials/kitties-part-01/9-conclusion.mdx:3]:
```suggestion
You have completed the Substrate Kitties Tutorial. You have learned how to build a blockchain using Substrate and how to create a runtime 
```

File: tutorials/kitties-part-01/9-conclusion.mdx [hunk 39]
```diff
@@ -0,0 +1,15 @@
+# Conclusion
+
+We have completed the Substrate Kitties Tutorial. We have learned how to build a blockchain using Substrate and how to create a runtime 
+module that implements the logic of our blockchain. We have also learned how to create a client-side application that interacts with our
```
Comment (niklasp) [inline tutorials/kitties-part-01/9-conclusion.mdx:4]:
```suggestion
module that implements the logic of our blockchain. You have also learned how to create a client-side application that interacts with your
```

**Review by niklasp — COMMENTED** [review]
Thank you Walquer, this is an outstanding tutorial and will be a great learning resource for the Polkadot developer community in my opinion. 

I made some small comments and changes, please address them, when finished, notify me and I will merge it!
